js/assessment.js

```diff
@@ -79,15 +79,18 @@ define([
                 reset completes before routing completes
             */
 
-            Adapt.trigger('plugin:beginWait');
-
-            for (var i = 0, l = pageAssessmentModels.length; i < l; i++) {
-                var pageAssessmentModel = pageAssessmentModels[i];
-                pageAssessmentModel.reset(false, function() {
-                    // N.B. this callback is asynchronous so [i] may have been incremented
-                    if (i >= l - 1) Adapt.trigger('plugin:endWait');
-                });
-            }
+            Adapt.wait.for(function(end) {
+
+                for (var i = 0, l = pageAssessmentModels.length; i < l; i++) {
+                    var pageAssessmentModel = pageAssessmentModels[i];
+                    pageAssessmentModel.reset(false, function() {
+                        // N.B. this callback is asynchronous so [i] may have been incremented
+                        if (i < l - 1) return;
+                        end();
```

Comment:
@chris-steele 
is it possible here for the last assessment to reset before the first, leaving i at l - 1 even though only the last assessment have been reset by this point? should i perhaps do a completed count just to make this clearer?

Reply:
@oliverfoster I'd say if the callback was asynchronous then yes, as we can't be sure of order, but the callback passed to reset is not used asychronously, unless I'm missing something? That code/comment must be a relic.

Reply:
@oliverfoster Yes I think you're right! We can't be sure of order with deferred calls can we, so we should amend this.

Reply:
(your first response is probably quite accurate) in practice i think we can assume that they're executed in order because setTimeout(cb,0) are queued in call order and called in queue order... but if we assume we know nothing about the asynchronous callback process of ``pageAssessmentModel.reset`` we can't be sure if each function might defer twice, or have extra defers if they're especially long, which would screw up the callbacks.

i'll change just in case.

Reply:
@oliverfoster yes that's much clearer, thanks